transform/app/services/solr/validate.py

```diff
@@ -1,17 +1,67 @@
+# pylint: disable=line-too-long, invalid-name, logging-fstring-interpolation
+"""Validate Solr"""
+import json
 import logging
+from typing import List
+
+import requests
 from fastapi import HTTPException
 from requests import RequestException
-from typing import List
+from requests.exceptions import ConnectionError as ReqConnectionError
 
+from app.services.solr.collections import get_collection_names, get_pined_collections
 from app.services.solr.configs import get_config_names
-from app.services.solr.collections import (
-    get_collection_names,
-    get_pined_collections,
-)
+from app.settings import settings
+from app.services.solr.utils import ids_mapping
 
 logger = logging.getLogger(__name__)
 
 
+def check_document_exists(
+    col_name: str,
+    data: dict | list[dict],
+) -> bool:
+    """Check if a solr document exists in all specified collections based on its ID"""
+    raw_id = data[0]["id"] if col_name == "interoperability guideline" else data["id"]
+    id_to_check = ids_mapping(raw_id, col_name)
```

Comment:
ids_mapping is not necessary here. IDs of IG and trainings are not mapped

Reply:
actually, it also should be generic and handle all data types. So mapping should be here. So lets keep it as it is now. Later on we will more generalize